modules/api/CMakeLists.txt

```diff
@@ -1,15 +1,23 @@
 cmake_minimum_required(VERSION 3.0.0)
 project(vmsapi VERSION 0.1.0)
 
+# Set default compile flags for GCC
+if(CMAKE_COMPILER_IS_GNUCXX)
+    message(STATUS "GCC detected, adding compile flags")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++98 -pedantic -Wall -Wextra")
+endif(CMAKE_COMPILER_IS_GNUCXX)
+
 # ----------------------------------------------------------------------------
 # Detect dependencies
 # ----------------------------------------------------------------------------
+option(CMAKE_THREAD_LIBS_INIT "thread library" ON)
+option(CMAKE_USE_WIN32_THREADS_INIT "using WIN32 threads" ON)
```

Comment:
Specify windows specific option only if the build platform is windows.

Example:
```
option(CMAKE_USE_WIN32_THREADS_INIT "using WIN32 threads" WIN32)
```